content/300-guides/150-testing/100-unit-testing.mdx

```diff
@@ -0,0 +1,311 @@
+---
+title: 'Unit testing'
+metaTitle: 'Unit testing with Prisma'
+metaDescription: 'Learn how to setup and run unit tests with the Prisma client'
+tocDepth: 3
+---
+
+<TopBlock>
+
+Unit testing aims to isolate a small portion (unit) of code and test it for logically predictable behaviors. It generally involves mocking objects or server responses to simulate real world behaviors. Some benefits to unit testing include:
+
+- Quickly find and isolate bugs in code.
+- Provides documentation for each module of code by way of indicating what certain code blocks should be doing.
+- A helpful gauge that a refactor has gone well. The tests should still pass after code has been refactored.
+
+In the context of Prisma, this generally means testing a function which makes database calls using the Prisma client.
+
+A single test should focus on the data returned by the Prisma client and how the function handles that data, ensuring that it behaves as expected.
```

Comment:
```suggestion
A single test should focus on the data returned by the Prisma Client and how the function handles that data, ensuring that it behaves as expected.
```

Reply:
Maybe a bit more emphasis on "A single test should focus on how **your function logic** handles different inputs (such as a `null` value or an empty list)."